src/core/atomic.d

```diff
@@ -375,6 +396,72 @@ else version (AsmX86_32)
         }
     }
 
+    shared(T) atomicExchange(MemoryOrder ms = MemoryOrder.seq,T,V)( shared(T)* here, V exchangeWith ) pure nothrow @nogc @safe
+        if ( !is(T == class) && !is(T U : U*) && __traits( compiles, { *here = exchangeWith; } ) )
```

Comment:
Just to make sure you've done your homework, are there any invalid memory orders for this operation? (I think I know the answer without checking anyway).

Reply:
There are no invalid orders.
Infact, the order does not affect codegen in any way; it should be routed through to the backend to inform code mobility.